packages/parser/src/parser.ts

```diff
@@ -187,4 +186,6 @@ function parseForESLint(
   return { ast, services, scopeManager, visitorKeys };
 }
 
-export { parse, parseForESLint, ParserOptions };
+export { parse, parseForESLint };
+
+export { ParserOptions } from '@typescript-eslint/types';
```

Comment:
```suggestion
export type { ParserOptions } from '@typescript-eslint/types';
```

Do we have [`consistent-type-exports`](https://typescript-eslint.io/rules/consistent-type-exports/) turned on? If not -- we should turn it on as part of this PR to ensure we're being strict.